app/forms/schools/add_participant_form.rb

```diff
@@ -6,20 +6,55 @@ class AddParticipantForm
     include ActiveModel::Attributes
     include ActiveModel::Serialization
 
+    STEPS = %i[type details not_implemented].freeze
+
     TYPE_OPTIONS = {
       ect: "Early Career Teacher",
       mentor: "Mentor",
     }.freeze
 
+    attribute :completed_steps, default: []
     attribute :type
+    attribute :full_name
+    attribute :email
 
     validates :type,
               on: :type,
               presence: { message: "Please select type of the new participant" },
               inclusion: { in: TYPE_OPTIONS.keys.map(&:to_s), allow_blank: true }
 
+    validates :full_name,
+              on: :details,
+              presence: true
+
+    validates :email,
+              on: :details,
+              presence: true,
+              format: { with: Devise.email_regexp, allow_blank: true }
+
     def type_options
       TYPE_OPTIONS
     end
+
+    def previous_step(current_step)
+      return completed_steps.last unless current_step.in?(completed_steps)
+
+      completed_steps[completed_steps.index(current_step) - 1]
+    end
+
+    def next_step(step = completed_steps.last)
+      case step
+      when :type then :details
+      else :not_implemented
+      end
+    end
```

Comment:
This method defines multistep form flow - it is very simple but will get more and more complex when we add more steps and possible forks. I **think** it is fine for now, however I am aware it violates open-closed principle. Ideally this would be relaying on polymorphism of the step objects, so it would be nice to revisit it / refactor when the MultistepForm is extracted.

Reply:
I don't think we want to spend time on writing a form wizard. If we think we need it let's use an existing solution - there are quite a few gems that do it around.

Reply:
The problem with existing solutions is that they do not allow dynamic step configuration (at least all the ones I've seen so far, maybe there is a diamond somewhere that hasn't pick much popularity yet).